robocop/checkers/misc.py

```diff
@@ -243,3 +243,20 @@ def visit_LibraryImport(self, node):  # noqa
         if not node.name:
             return
         self.libraries.append(node)
+
+
+class EmptyVariableChecker(VisitorChecker):
+    """ Checker for variables without value. """
+    rules = {
+        "0912": (
+            "empty-variable",
+            "Use built-in variable ${EMPTY} instead of leaving variable without value",
+            RuleSeverity.INFO
+        )
+    }
+
+    def visit_Variable(self, node):  # noqa
+        if not node.name:
+            return
+        if not node.value or all(not val for val in node.value):
```

Comment:
It will only raport those variables that have all values empty, ie:

```
@{ARRAY}
...
...
```
but it will not catch mixed empty:
```
@{ARRAY}
...
...  value
```
It will also not catch ``MULTILINE_WITH_EMPTY_LINES`` from your test data - it's actually array of two values, one empty and second ${EMPTY}, with comment in between them. It should be catched.

The easiest way of achieving this would be problably calling get_tokens on variable looking for Token.Argument and check them one by one (it's better to check on tokens because then you can point to exact place where there is empty value). So in this case:
```
@{ARRAY}
...  1
...  2
...  3
...  a
...  b
...  c
...
...  5
```
It will only raport on ``...  `` and not whole variable